tedana/decomposition/ica.py

```diff
@@ -50,6 +53,51 @@ def tedica(data, n_components, fixed_seed, maxit=500, maxrestart=10):
         "decompose the dimensionally reduced dataset."
     )
 
+    if ica_method=='robustica':
+        mmix, Iq = r_ica(data, n_components, n_robust_runs, maxit)
+        fixed_seed=-99999
+    elif ica_method=='fastica':
+        mmix, fixed_seed=f_ica(data, n_components, fixed_seed, maxit=500, maxrestart=10)
+        Iq = 0
+    else:
+        LGR.warning("The selected ICA method is invalid!")
+        sys.exit()
+
+
+
+
+    return mmix, fixed_seed
+
+
+def r_ica(data, n_components, n_robust_runs, max_it): ####BTBTBTB:
+
+    if n_robust_runs>100:
+        LGR.warning("The selected n_robust_runs is a very big number!")
```

Comment:
Don't know the extent to which this is arbitrary and/or the right way to go about this.

Think about the actual purpose here. Ultimately what you're trying to achieve is that if the user has specified a very large number, either by accident or because they don't understand the function of the parameter, the software may take an unwieldy time to complete, and therefore you would like for there to be some form of foreshadowing of this fact so that the user may choose to abort immediately.

Two points then. Firstly, "this number is a big number!" doesn't necessarily communicate this, especially in the case where a user has made that mistake specifically because they don't know what the parameter does. An up-front message could be more specific about an expectation of unreasonable execution time. Secondly, you could think about alternative ways of giving the user the information they need. For instance, with RobustICA, is there any way to generate a progress bar or the like?

Reply:
It is updated to 

> LGR.warning("The selected n_robust_runs is a very big number! The process will take a long time!")

Moreover, in the CLI choices I limited it to "choices=range(5,500)".

The robustica is set such that it shows the progress bar and a dynamic estimation of each iteration.


![image](https://github.com/BahmanTahayori/tedana_florey/assets/19523962/966473bf-eed1-4581-b254-56b45005e85f)